### DISCUSSION TITLE
How to access Cloudflare bindings in server functions?

### DISCUSSION THREAD

**Discussion by asterikx** [solidjs/solid-start#1197] [Q&A]
As Cloudflare bindings (KV, D1, ...) are only available at runtime, how can I access them inside server functions in Beta 2?

**Comment by boehs** [comment]
Have you tried just using them according to the cloudflare docs? That's what I do and it works fine. If you have a more specific issue, I'd encourage you to ask that instead.

**Reply by RonenEizen** [reply]
@boehs Can you explain further? How do you access it in the following example?

```typescript
import { For } from "solid-js";
import { createAsync, cache } from "@solidjs/router";
 
const getData = cache(async () => {
  // I want to access something from D1 or KV here and return it
}, "data");
 
export const route = {
  load: () => getData()
};
 
export default function Page() {
  const data = createAsync(() => getData());
 
  return <div>{data()}</div>;
}
```
